src/main/java/com/cefriel/template/io/sql/SQLReader.java

```diff
@@ -35,20 +37,39 @@ public class SQLReader implements Reader {
 
     private Connection conn;
 
+    private List<String> tables = new ArrayList<>();
+
     private String queryHeader;
     private boolean verbose;
     private static final Object lock = new Object();
 
 
-    public SQLReader(String driver,  String url, String database, String username, String password) {
+    public SQLReader(String driver, String url, String database, String username, String password) {
         if (!url.contains("jdbc:"))
             url = "jdbc:" + driver + "://" + url + "/" + database;
         log.info("Connection to database with URL: " + url);
 
         try {
             conn = DriverManager.getConnection(url, username, password);
+            PreparedStatement tablesQueryStatement;
```

Comment:
Great work! Please split the table names retrieval in a separate function and invoke it also for the other constructor (i.e., in the case a Connection is directly provided)

Reply:
adressed in 8a23da9